.travis.yml

```diff
@@ -1,38 +1,48 @@
 dist: trusty
 sudo: false
 language: cpp
+os: linux
 
 matrix:
   include:
     # GCC 4.9
-    - os: linux
+    - env: COMPILER=g++-4.9
       addons:
         apt:
           sources:
             - ubuntu-toolchain-r-test
           packages:
             - g++-4.9
-      env: MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
 
     # Clang 3.8
-    - os: linux
+    - env: COMPILER=clang++-3.8
       addons:
         apt:
           sources:
             - ubuntu-toolchain-r-test
             - llvm-toolchain-precise-3.8
           packages:
             - clang-3.8
-      env: MATRIX_EVAL="CC=clang-3.8 && CXX=clang++-3.8"
+
+    # Clang 7 (default on Xenial)
+    - env: COMPILER=clang++-7
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-trusty-7
+          packages:
+            - clang-7
 
 before_install:
   - sudo add-apt-repository -y ppa:samuel-bachmann/boost
   - sudo apt-get -y -d update
-  - eval "${MATRIX_EVAL}"
 install:
-  - sudo apt-get -y install cmake3 libboost-test1.60-dev libboost-date-time1.60-dev
+  - sudo apt-get -y install cmake3 ninja-build libboost-test1.60-dev libboost-date-time1.60-dev
+  - export CXX=${COMPILER}
+  - ${CXX} --version
```

Comment:
Good idea.

Huh, it looks like the compiler settings don't take. See output for:
  * [g++ 4.9](https://travis-ci.org/exercism/cpp/jobs/505040922#L587) => `g++ 4.8.4`
  * [clang++ 3.8](https://travis-ci.org/exercism/cpp/jobs/505040923#L591) => `g++ 4.8.5`
  * [clang++ 7](https://travis-ci.org/exercism/cpp/jobs/505040924#L629) => `g++ 4.8.4`

I expected to see something like this for Clang (from my personal Linux box):
```bash
$ clang++-3.8 --version
clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)
```

Reply:
I see - I was looking for the output in the wrong place.

Reply:
Yea, the travis output takes a bit of getting used to, it's not terribly obvious.